src/Fixer/ClassNotation/OrderedImplementsFixer.php

```diff
@@ -0,0 +1,208 @@
+                    [self::OPTION_DIRECTION => self::DIRECTION_DESCEND]
+                ),
+                new CodeSample(
+                    "<?php\n\nfinal class Example implements MuchLonger, Short, Longer {}\n",
+                    [self::OPTION_ORDER => self::ORDER_LENGTH]
+                ),
+                new CodeSample(
+                    "<?php\n\nfinal class Example implements MuchLonger, Short, Longer {}\n",
+                    [
+                        self::OPTION_ORDER => self::ORDER_LENGTH,
+                        self::OPTION_DIRECTION => self::DIRECTION_DESCEND,
+                    ]
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isTokenKindFound(T_IMPLEMENTS);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        foreach ($tokens as $index => $token) {
+            if (!$token->isGivenKind(T_IMPLEMENTS)) {
+                continue;
+            }
+
+            $interfaceIndex = 0;
+            $interfaces = [['tokens' => []]];
+
+            $implementsStart = $index + 1;
+            $classStart = $tokens->getNextTokenOfKind($implementsStart, ['{']);
+            $implementsEnd = $tokens->getPrevNonWhitespace($classStart);
+
+            for ($i = $implementsStart; $i <= $implementsEnd; ++$i) {
+                if ($tokens[$i]->equals(',')) {
+                    ++$interfaceIndex;
+                    $interfaces[$interfaceIndex] = ['tokens' => []];
+
+                    continue;
+                }
+
+                $interfaces[$interfaceIndex]['tokens'][] = $tokens[$i];
+            }
+
+            if (1 === \count($interfaces)) {
+                continue;
+            }
+
+            foreach ($interfaces as $interfaceIndex => $interface) {
+                $interfaceTokens = Tokens::fromArray($interface['tokens'], false);
+
+                $normalized = '';
+                $actualInterfaceIndex = $interfaceTokens->getNextMeaningfulToken(-1);
+
+                while ($interfaceTokens->offsetExists($actualInterfaceIndex)) {
+                    $token = $interfaceTokens[$actualInterfaceIndex];
+
+                    if (null === $token || $token->isComment() || $token->isWhitespace()) {
+                        break;
+                    }
+
+                    $normalized .= str_replace('\\', ' ', $token->getContent());
+                    ++$actualInterfaceIndex;
+                }
+
+                $interfaces[$interfaceIndex]['normalized'] = $normalized;
+                $interfaces[$interfaceIndex]['originalIndex'] = $interfaceIndex;
+            }
+
+            usort($interfaces, function (array $first, array $second) {
+                $score = self::ORDER_LENGTH === $this->configuration[self::OPTION_ORDER]
```

Comment:
if sorting on length on normalized data, would '\A\B\C' not considered less of length than 'ABCD' ?

Reply:
If you're looking at actual length, `A\B\C\D` **is longer** than `Abcde` so it should be marked as longer, normalised or not.

Also, I looked at `OrderedImportsFixer` for inspiration, which handles it the same.